### ISSUE TITLE
Trouble running on Windows -- $(npm bin)

### ISSUE THREAD

**Issue by alexisgo** [Automattic/google-docs-add-on#88]
Hello to you fine folks,

I'm new to npm and node, so apologies if this is straghtforward, but after installing npm and cloning this project, as well as taking all the steps described in the README, this is failing for me with `npm run deploy` with the following error:

'$' is not recognized as an internal or external command,
operable program or batch file.

npm ERR! Windows_NT 10.0.15063
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "deploy"
npm ERR! node v6.11.2
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! wordpress-for-google-docs@1.4.5 deploy: `npm run build && $(node bin) gapps upload`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the wordpress-for-google-docs@1.4.5 deploy script 'npm run build && $(node bin) gapps upload'.

Am I supposed to be editing line 13 in package.json to something else? I mean this line specifically:
    `"deploy": "npm run build && $(npm bin)/gapps upload",`

**Comment by alexisgo** [comment]
I was able to get this working by changing line 13 in package.json to `"deploy": "npm run build && $(npm bin)/gapps upload"` FWIW ... just trying to understand what $(npm bin) does in theory, so still interested in the answer to this if anyone has the time.

Thank you!

**Comment by georgeh** [comment]
tl;dr: `$(npm bin)/gapps upload` runs the [node-google-app-script](https://github.com/danthareja/node-google-apps-script) uploader.

[`npm bin`](https://docs.npmjs.com/cli/bin) expands to the path where npm packages install their executables (e.g. `/home/you/google-docs-add-on/node_modules/.bin`)

`$(something)` tells bash to use the output of `something` as part of the command. In this case, I'm using it to get the absolute path to `gapps` because I don't count on it being installed globally. Theres a really good chance `$(npm bin)/gapps upload` is the same as `node_modules/.bin/gapps upload`. I went with `$(npm bin)` for using an absolute path at the expense of being cleverer.

I haven't tested deploying on Windows, but if you got it working then please submit a PR

**Comment by originAsad** [comment]
Just use after the npm bin command it will gave the same output work in my laptop in windows
